fix(helm): Init not creating local-index symlink on Windows

pietromenna opened the pull request:
os.Symlink required additional permissions on Windows, and init is not currently identifying the failed creation.

Related to #2071

bacongobbler reviewed (changes requested):
Left a comment on Slack about how we should approach this with conditional build tags: https://dave.cheney.net/2013/10/12/how-to-use-conditional-compilation-with-the-go-build-tool

pietromenna committed:
fix(helm): Init not creating local-index symlink on Windows

os.Symlink required additional permissions on Windows, and init is not currently identifying the failed creation.

Related to #2071

bacongobbler commented on cmd/helm/init_unix.go:
```diff
@@ -0,0 +1,27 @@
+// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
```
you can change this to `// +build !windows` to simplify this.

bacongobbler commented on cmd/helm/init_windows.go:
```diff
@@ -0,0 +1,25 @@
+/*
```
You'll need to add a build tag here e.g. `// +build windows`

pietromenna commented:
Hi @bacongobbler .

I incorporated the changes as you suggested.

bacongobbler commented:
The CI failure was something we fixed 2 weeks ago and isn't relevant to this PR. Merging!
